p4_constraints/frontend/BUILD.bazel

```diff
@@ -121,5 +121,5 @@ cc_library(
 cc_library(
     name = "constraint_kind",
     hdrs = ["constraint_kind.h"],
-    visibility = ["//p4_constraints/backend:__pkg__"],
+    visibility = ["//visibility:public"],
```

Comment:
My preference would be for `p4_constraints:__subpackages__`, if that is sufficient?

Reply:
From what I understand this would not extend the visibility to `p4lang_p4c`, right? I could add this one instead considering it is defined in the Bazel deps?

Reply:
I don't think that will work from the persepctive of p4c, have you tried?

If we need it to be accessible outside of this repo, making it public is fine, no concerns.